agent/xds/clusters.go

```diff
@@ -385,6 +413,49 @@ func (s *ResourceGenerator) makeGatewayServiceClusters(
 	return clusters, nil
 }
 
+func (s *ResourceGenerator) makeDestinationClusters(cfgSnap *proxycfg.ConfigSnapshot) ([]proto.Message, error) {
+	var createDynamicForwardProxy bool
+	serviceConfigs := cfgSnap.TerminatingGateway.ServiceConfigs
+
+	clusters := make([]proto.Message, 0, len(cfgSnap.TerminatingGateway.DestinationServices))
+
+	for _, svcName := range cfgSnap.TerminatingGateway.ValidDestinations() {
+		svcConfig, ok := serviceConfigs[svcName]
```

Comment:
Yes. I kept the check here in case there were any weird inconsistency issues with the serviceConfigs updating in the background, but if you don't think that's an issue, I'm happy to remove.

Reply:
All of this run in the same thread so I don't think this check could ever happen. I think it's safe to remove it.